vortex-alp/src/compress.rs

```diff
@@ -166,28 +169,38 @@ mod tests {
         let encoded = alp_encode(&array).unwrap();
         assert!(encoded.patches().is_none());
         assert_eq!(
-            encoded.encoded().as_primitive().typed_data::<i32>(),
+            encoded
+                .encoded()
+                .clone()
+                .flatten_primitive()
```

Comment:
Downcast, don't flatten

Reply:
Will replace with PrimitiveArray::try_from downcast -- still need to clone though I think

Reply:
Ooh, replacing this causes a test to fail 0_o